frontend/test/simulate/frontend_test.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2025 Microsoft Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package simulate
+
+import (
+	"context"
+	"encoding/json"
+	"io/fs"
+	"os"
+	"os/signal"
+	"syscall"
+	"testing"
+
+	csarhcpv1alpha1 "github.com/openshift-online/ocm-api-model/clientapi/arohcp/v1alpha1"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+
+	"github.com/Azure/ARO-HCP/internal/api"
+	"github.com/Azure/ARO-HCP/internal/ocm"
+)
+
+func TestFrontendClusterRead(t *testing.T) {
+	if os.Getenv("FRONTEND_SIMULATION_TESTING") != "true" {
+		t.Skip("Skipping test")
+	}
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	frontend, testInfo, err := NewFrontendFromTestingEnv(ctx, t)
+	require.NoError(t, err)
+	defer testInfo.Cleanup(context.Background())
+
+	stop := make(chan struct{})
+	signalChannel := make(chan os.Signal, 1)
+	signal.Notify(signalChannel, syscall.SIGINT, syscall.SIGTERM)
+	go frontend.Run(ctx, stop)
+
+	err = testInfo.CreateInitialCosmosContent(ctx, api.Must(fs.Sub(artifacts, "artifacts/ClusterReadOldData/initial-cosmos-state")))
+	require.NoError(t, err)
+
+	clusterServiceCluster, err := csarhcpv1alpha1.UnmarshalCluster(api.Must(artifacts.ReadFile("artifacts/ClusterReadOldData/initial-cluster-service-state/02-some-cluster.json")))
+	require.NoError(t, err)
+	testInfo.MockClusterServiceClient.EXPECT().GetCluster(gomock.Any(), api.Must(ocm.NewInternalID("/api/clusters_mgmt/v1/clusters/fixed-value"))).Return(clusterServiceCluster, nil)
```

Comment:
nit for the future:
We should use `aro_hcp/v1alpha1` instead of `clusters_mgmt/v1` here and in test data.  Only because `ocm.NewInternalID` will likely stop recognizing the `clusters_mgmt` API in the foreseeable future and these tests will break when that day comes.